### PR TITLE
Fix dependencies on test modules

### PR DESCRIPTION
**Pull request by renatocaval**
This fix and clean-ups the dependencies on the test modules.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -15,27 +15,27 @@ lazy val coreTest =
   Project(id = "akka-projection-core-test", base = file("akka-projection-core-test"))
     .configs(IntegrationTest)
     .settings(Defaults.itSettings)
-    .settings(Protobuf.settings)
```
Comment (renatocaval) [inline build.sbt:18]: This was not needed. Probably a copy-n-paste from another module.

File: build.sbt [hunk 2]
```diff
@@ -15,27 +15,27 @@ lazy val coreTest =
...
+    .settings(Dependencies.coreTest)
     .settings(publish / skip := true)
-    .dependsOn(core % "compile->compile;test->test")
-    .dependsOn(testkit % "test->test")
+    .dependsOn(core)
+    .dependsOn(testkit)
```
Comment (renatocaval) [inline build.sbt]: It depends on the core and don't use the test source from core.

File: build.sbt [hunk 3]
```diff
@@ -15,27 +15,27 @@ lazy val coreTest =
...
+    .dependsOn(core)
 
 // provides offset storage backed by a JDBC table
 lazy val jdbc =
   Project(id = "akka-projection-jdbc", base = file("akka-projection-jdbc"))
     .configs(IntegrationTest.extend(Test))
     .settings(Defaults.itSettings)
     .settings(Dependencies.jdbc)
-    .dependsOn(core % "compile->compile")
+    .dependsOn(core)
```
Comment (renatocaval) [inline build.sbt:36]: `compile->compile` is the default, no need to repeat it.

File: build.sbt [hunk 4]
```diff
@@ -15,27 +15,27 @@ lazy val coreTest =
   Project(id = "akka-projection-core-test", base = file("akka-projection-core-test"))
     .configs(IntegrationTest)
     .settings(Defaults.itSettings)
-    .settings(Protobuf.settings)
+    .settings(Dependencies.coreTest)
     .settings(publish / skip := true)
-    .dependsOn(core % "compile->compile;test->test")
-    .dependsOn(testkit % "test->test")
+    .dependsOn(core)
```
Comment (renatocaval) [inline build.sbt:20]: core-test depends on core, nothing else

File: build.sbt [hunk 5]
```diff
@@ -15,27 +15,27 @@ lazy val coreTest =
...
+    .dependsOn(testkit % "test")
```
Comment (renatocaval) [inline build.sbt:21]: `testkit` is a library. In core-test we use it in test scope. We were doing it transitively vie `testkit` tests scope, that's not correct.

File: build.sbt [hunk 6]
```diff
@@ -15,27 +15,27 @@ lazy val coreTest =
...
 
 lazy val testkit =
   Project(id = "akka-projection-testkit", base = file("akka-projection-testkit"))
     .configs(IntegrationTest)
     .settings(Defaults.itSettings)
     .settings(Dependencies.testKit)
-    .dependsOn(core % "compile->compile;test->test")
+    .dependsOn(core)
```
Comment (renatocaval) [inline build.sbt:28]: testkit doesn't depend on the tests in core

File: build.sbt [hunk 7]
```diff
@@ -55,15 +55,15 @@ lazy val cassandra =
     .settings(Defaults.itSettings)
     .settings(Dependencies.cassandra)
     .dependsOn(core)
-    .dependsOn(coreTest % "test->compile;it->test")
+    .dependsOn(coreTest % "test->test;it->test")
```
Comment (renatocaval) [inline build.sbt:61]: Here is a special case. Cassandra project doesn't have test sources, only integration tests.  Without adding `test->test`, the pom.xml don't get it declared.

**Review by renatocaval — COMMENTED** [review]
left a few comments to clarify

**Review by ennru — APPROVED** [review]
Please mark scopes with `Test` instead of `"test"`.

LGTM.

**Comment by renatocaval** [comment]
Maybe some extra information that I discovered while testing this new build. 

Module `core-test` is never published and is only used in test or it:test scopes. 

When a module depends on `core-test` and defined the scope as `test->test`, meaning the test sources from core-test are available in test scope, then the pom for that module will have the following deps:

```xml
<dependency>
  <groupId>com.lightbend.akka</groupId>
  <artifactId>akka-projection-core-test_2.13</artifactId>
  <version>1.0.0-RC2</version>
  <scope>test</scope>
</dependency>
```

If the module has the scope defined as `it->test`, the generate pom won't have the entry.

To keep things aligned, I added `test->test` to the cassandra module (see my comment [here](https://github.com/akka/akka-projection/pull/412/files#r480096861)).

I'm not familiar with test scope in artifact pom.xml. I hope that this is only informative because core-test is never published.

**Review by ignasi35 — APPROVED** [review]
Nice. Cleaner and better documented (cassandra).

**Comment by ignasi35** [comment]
> use Test scope instead

👍🏼

**Pull request merged**